lib/puppet/provider/nssdb/certutil.rb

```diff
@@ -0,0 +1,44 @@
+Puppet::Type.type(:nssdb).provide(:certutil) do
+  commands :certutil => 'certutil'
+
+  def create
+    generate_nssdb
+  end
+
+  def destroy
+    delete_nssdb
+  end
+
+  def exists?
```

Comment:
Instead of checking that certain files exist (cause that has constraints as we see in todays code having ti know the full set of files per platform) this could use native methods in `certutil` to verify a working database exists:

```
# certutil -L -d /etc/pki/katello/nssdb/

Certificate Nickname                                         Trust Attributes
                                                             SSL,S/MIME,JAR/XPI

ca                                                           CT,C,c
broker                                                       u,u,u

# certutil -L -d /etc/pki/katello/
certutil: function failed: SEC_ERROR_LEGACY_DATABASE: The certificate/key database is in an old, unsupported format.
```

Reply:
Ok, so I have the implementation

```ruby
  def exists?
    certutil(
      '-L',
      '-d', resource[:nssdb_dir]
    )
  end
```

I think it's not necessary to `rescue Puppet::ExecutionFailure` assuming the `commands` method handles exit codes the way I think it should (but I couldn't find that documented).

Reply:
I would think you'd want the rescue to convert the error into a debug message and move on. The error generated if the NSS database does not exist should be indicator we need to create it not a blocking failure. I think this is shown in the test output:

```
  Error: /Stage[main]/Certs::Ssltools::Nssdb/Nssdb[/etc/pki/katello/nssdb]: Could not evaluate: Execution of '/usr/bin/certutil -L -d /etc/pki/katello/nssdb' returned 255: certutil: function failed: SEC_ERROR_LEGACY_DATABASE: The certificate/key database is in an old, unsupported format.
```